### PR TITLE
dist/tools/compile_commands: fix clangd mode with ESP32

### PR DESCRIPTION
**Pull request by maribu**
### Contribution description

In clangd mode drop a number of compiler flags not supported by LLVM.

### Testing procedure

Run `make compile-commands BOARD=<SOME-ESP32-board> -C examples/default` and open a C file with an editor that uses clangd as language server (e.g. vim/neovim with ALE, or VS Code). In `master` clangd should be really unhappy, with this it should provide decent linting/analysis/code completion/....

### Issues/PRs references

None

### CODE REVIEW

File: dist/tools/compile_commands/compile_commands.py [hunk 1]
```diff
@@ -318,11 +318,21 @@ def generate_compile_commands(args):
...
+            '-msmall-data-limit=8',
+            # not supported by clang, see
+            # https://gcc.gnu.org/onlinedocs/gcc-10.2.0/gcc/Xtensa-Options.html
+            '-mtext-section-literals',
+            '-fstrict-volatile-bitfields',
+            # it's called -mlong-calls in LLVM, but we don't need it for clangd
+            # as we do not generate code anyway
+            '-mlongcalls',
+        ]
+        _args.filter_out.extend(flags)
```
Comment (maribu) [inline dist/tools/compile_commands/compile_commands.py:337]: I think this also fixes a bug drive-by, as previously arguments provided by `--filter-out` were overwritten when using `--clangd`, rather than extended.

**Review by gschorcht — APPROVED** [review]
